src/Orleans/Lifecycle/LifecycleObservable.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans
+{
+    public class LifecycleObservable<TStage> : ILifecycleObservable<TStage>, ILifecycleObserver
```

Comment:
Orleans users: probably not.
Maintainers may need to create different instances if lifecycles have special needs, but I wouldn't call it expected, more like possible.